stream_framer.go

```diff
@@ -110,13 +109,16 @@ func (f *streamFramer) maybePopNormalFrames(maxBytes protocol.ByteCount) (res []
 		frame.Offset = s.GetWriteOffset()
 		// not perfect, but thread-safe since writeOffset is only written when getting data
 		frameHeaderBytes, _ := frame.MinLength(f.version) // can never error
-		if currentLen+frameHeaderBytes > maxBytes {
+		if currentLen+frameHeaderBytes > maxTotalLen {
 			return false, nil // theoretically, we could find another stream that fits, but this is quite unlikely, so we stop here
 		}
-		maxLen := maxBytes - currentLen - frameHeaderBytes
+		maxLen := maxTotalLen - currentLen
+		if maxLen < protocol.MinStreamFrameSize { // don't try to add new STREAM frames, if only little space is left in the packet
```

Comment:
Maybe same as above, but I'll leave that up to you.

Reply:
The idea here is to exit from the callback early, since ranging over the streams is expensive.
As long as the majority of `Write`s are larger than the packet size, we won't often pack a packet that is smaller than the maximum packet size.